modules/battery_manager/ajax/add_entry.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Battery Manager adder
+ *
+ * Checks Test Battery for duplicates and handles insertions into Test Battery
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Battery_Manager
+ * @author   Victoria Foing <victoria.foing@mcin.ca>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+
+
+// Determine which action is called
+if (isset($_GET['action'])) {
+    $action = $_GET['action'];
+    if ($action == "checkForDuplicate") {
+        echo checkForDuplicate();
+    } else if ($action == "add") {
+        addEntry();
+    }
+}
+
+/**
+ * Return duplicate of entry if it exists and null otherwise
+ *
+ * @return Object or null
+ */
+function checkForDuplicate()
+{
+    $db =& \Database::singleton();
+
+    // Retrieve values entered by user
+    $form_data = getFormData();
+
+    // Build SQL query based on values entered by user
+    $query     = " SELECT * FROM test_battery ";
+    $i         = 0;
+    $connector = "WHERE ";
+    foreach ($form_data as $key => $value) {
```

Comment:
not sure about this. in my experience, an entry should be considered a duplicate if there is an existing entry based on `Test_name, Stage, SubprojectID, Visit_label`, some projects might also include `CenterID` as well. 
this would allow duplicates through variances in AgeMin/Max and instr_order